### PR TITLE
docs(datadog_metrics sink): use auto-generated config docs

### PR DESCRIPTION
**Pull request by neuronull**
closes: https://github.com/vectordotdev/vector/issues/16241

epic: https://github.com/vectordotdev/vector/issues/14999

### CODE REVIEW

File: src/sinks/datadog/metrics/config.rs [hunk 1]
```diff
@@ -92,37 +92,50 @@ impl DatadogMetricsEndpointConfiguration {
...
+    #[serde(default)]
     pub region: Option<Region>,
 
     /// The Datadog [site][dd_site] to send metrics to.
     ///
     /// [dd_site]: https://docs.datadoghq.com/getting_started/site
     #[serde(default = "default_site")]
+    #[configurable(metadata(docs::examples = "us3.datadoghq.com"))]
+    #[configurable(metadata(docs::examples = "datadoghq.eu"))]
     pub site: String,
```
Comment (spencergilbert) [inline src/sinks/datadog/metrics/config.rs:128]: **thought**: As the possible sites are known, they could be an enum to return improved user feedback on misconfigurations.

**Review by spencergilbert — COMMENTED** [review]
🤔 there wasn't a preview built? Can we take a quick poke to make sure this didn't somehow _break_ things?

File: src/sinks/datadog/metrics/config.rs (hunk 1 above)
Comment (neuronull) [reply src/sinks/datadog/metrics/config.rs:128]: The main issue I can conceive with that, is that if they add a new site, how would we know to update that enum 🤔 

To get us closer to that, I could add the other current sites as examples ?

Comment (spencergilbert) [reply src/sinks/datadog/metrics/config.rs:128]: Sure, maybe not necessary - lazy answer could be linking out to the DD docs that list all the sites (if that exists).

Definitely a non-blocking comment - just came up as I'd been thinking a lot about improving the user experience

Comment (neuronull) [reply src/sinks/datadog/metrics/config.rs:128]: I like the idea. I think I'll tackle that in the mini refactor, so don't have to edit each branch.

Comment (spencergilbert) [reply src/sinks/datadog/metrics/config.rs:128]: Sounds good to me!

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**